ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/attestation/AggregatingAttestationPool.java

```diff
@@ -269,16 +269,24 @@ public synchronized SszList<Attestation> getAttestationsForBlock(
 
   public synchronized List<Attestation> getAttestations(
       final Optional<UInt64> maybeSlot, final Optional<UInt64> maybeCommitteeIndex) {
+
     final Predicate<Map.Entry<UInt64, Set<Bytes>>> filterForSlot =
         (entry) -> maybeSlot.map(slot -> entry.getKey().equals(slot)).orElse(true);
 
-    // TODO fix for electra (only used in Beacon API)
     final Predicate<MatchingDataAttestationGroup> filterForCommitteeIndex =
         (group) ->
             maybeCommitteeIndex
                 .map(index -> group.getAttestationData().getIndex().equals(index))
                 .orElse(true);
 
+    final SchemaDefinitions schemaDefinitions = getSchemaDefinitions(maybeSlot);
+
+    final SszListSchema<Attestation, ?> attestationsSchema =
+        schemaDefinitions.getBeaconBlockBodySchema().getAttestationsSchema();
+
+    final boolean requestRequiresAttestationWithCommitteeBits =
+        schemaDefinitions.getAttestationSchema().requiresCommitteeBits();
+
```

Comment:
this seems like a lot of work to know we're in electra... 
We're getting schemadefinitions, possibly creating them every slot, all to just check we're in electra milestone or beyond, i think we might want to think about what we're doing here in a synchronized function that will be used a bunch.

Reply:
right, made the schema definitions retrieval synchronised and did the same for the already existing attestation for blocks

Reply:
I think at the very least we should probably be using the schema definition cache and avoiding potentially creating it all continually...

Reply:
ok i think we're ok here actually, its created as the specVersion is created. this should be ok.